specs/phase0/fork-choice.md

```diff
@@ -218,17 +267,30 @@ def filter_block_tree(store: Store, block_root: Root, blocks: Dict[Root, BeaconB
             return True
         return False
 
-    # If leaf block, check finalized/justified checkpoints as matching latest.
-    head_state = store.block_states[block_root]
+    current_epoch = compute_epoch_at_slot(get_current_slot(store))
+    voting_source = get_voting_source(store, block_root)
 
+    # The voting source should be at the same height as the store's justified checkpoint.
     correct_justified = (
         store.justified_checkpoint.epoch == GENESIS_EPOCH
-        or head_state.current_justified_checkpoint == store.justified_checkpoint
+        or voting_source.epoch == store.justified_checkpoint.epoch
     )
+
+    # If the block should be pulled-up due to previous epoch being justified, also check
+    # that the unrealized justification is higher than the store's justified
+    # checkpoint, and the voting source is not more than two epochs ago.
+    if not correct_justified and is_previous_epoch_justified(store):
+        correct_justified = (
+            store.unrealized_justifications[block_root].epoch >= store.justified_checkpoint.epoch and
+            voting_source.epoch + 2 >= current_epoch
```

Comment:
Consider adding a comment here.
and/or consider defining a var for  `2` to make this more descriptive